R/render.R

```diff
@@ -22,14 +22,23 @@ create_prevelance_renderer <- function(
   renderer
   ) {
   function(timestep) {
+    
     asymptomatic <- state$get_index_of('A')
-    prob <- probability_of_detection(
-      get_age(birth$get_values(asymptomatic), timestep),
-      immunity$get_values(asymptomatic),
-      parameters
-    )
-    asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
-
+    
+    if(parameters$parasite =="falciparum"){
+      prob <- probability_of_detection(
+        get_age(birth$get_values(asymptomatic), timestep),
+        immunity$get_values(asymptomatic),
+        parameters
+      )
+      asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
+      
+    } else if (parameters$parasite =="vivax"){
+      # The vivax model defines asymptomatic infections as being detectable by PCR
```

Comment:
But I think prevalence here is (as default) by microscopy? If so then fewer would be detected than via pcr

Reply:
See Phi_LM in the White et all (2018) SI

Reply:
Apologies - this was a typo. PCR should read as "detectable by light-microscopy".

Reply:
Perhaps this is an opportunity to make the columns more specific, e.g. with `n_detect_microscopy_` for falciparum and `n_detect_LM_` and `n_detect_PCR_` for vivax runs?

We would have to reflect that in the output documentation in `model.R:run_simulation` and maybe the model vignette.

Reply:
In the vivax model, the asymptomatic state is defined as being detectable by LM, while sub-patent is defined as being detectable by PCR, so those things are already rendered (unless I've misunderstood). Does that mean we should just remove n-detect for vivax...? Unless we want to add in some sort of probability of LM/PCR detection that isn't just all the individuals in that state...

Reply:
Yes, that makes sense! Let's remove n_detect.

Reply:
n_detect and p_detect for each parasite species revised now